docs/traffic-mirroring-proposal.md

```diff
@@ -0,0 +1,1282 @@
+            {
+                "type": "ovs",
+                "bridge": "br1",
+                "vlan": 300
+            }
+        ]
+    }
+---
+apiVersion: "k8s.cni.cncf.io/v1"
+kind: NetworkAttachmentDefinition
+metadata:
+  name: ovs-cons-mir
+spec:
+  config: |-
+    {
+        "cniVersion": "0.4.0",
+        "plugins": [
+            {
+                "type": "ovs",
+                 "bridge": "br1"
+            },
+            {
+                "type": "ovs-mirror",
+                 "bridge": "br1"
+            }
+        ]
+    }
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: ovs-prod-1
+spec:
+  template:
+    metadata:
+      annotations:
+        k8s.v1.cni.cncf.io/networks: |-
+          [
+            {
+                "name":"ovs-vlan100-mir",
+                "interface":"eth1",
+                "namespace":"emu-cni",
+                "cni-args": {
+                    "mirrors": [
+                        {
+                            "name": "mirror-1",
+                            "selection": ["src-port"]
+                        }
+                    ]
+                }
+            }
+          ]
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: ovs-prod-2
+spec:
+  template:
+    metadata:
+      annotations:
+        k8s.v1.cni.cncf.io/networks: |-
+          [
+            {
+                "name":"ovs-vlan200-mir",
+                "interface":"eth1",
+                "namespace":"emu-cni",
+                "cni-args": {
+                    "mirrors": [
+                        {
+                            "name": "mirror-2",
+                            "selection": ["src-port"]
+                        }
+                    ]
+                }
+            }
+          ]
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: ovs-cons-1
+spec:
+  template:
+    metadata:
+      annotations:
+        k8s.v1.cni.cncf.io/networks: |-
+          [
+            {
+                "name":"ovs-vlan300",
+                "interface":"eth1",
+                "namespace":"emu-cni"
+            },
+            {
+                "name":"ovs-cons-mir",
+                "interface":"eth2",
+                "namespace":"emu-cni",
+                "cni-args": {
+                    "mirrors": [
+                        {
+                            "name": "mirror-1",
+                            "output": "port"
+                        }
+                    ]
+                }
+            },
+            {
+                "name":"ovs-cons-mir",
+                "interface":"eth3",
+                "namespace":"emu-cni",
+                "cni-args": {
+                    "mirrors": [
+                        {
+                            "name": "mirror-2",
+                            "output": "port"
+                        }
+                    ]
+                }
+            }
+          ]
+```
+
+### Pros and Cons - NAD approach VS cni-args approach
```

Comment:
Security is the highest priority, user experience is close second though.

Would it make sense to start by configuring this through the NAD. And as a follow-up introduce a parameter `allowPodArgs` (or something similar) that would enable users to configure the mirror through parameters on pods? That way the admin has full control and option to loosen the security.

Reply:
Totally agree with you about security!
Your idea seems very reasonable, just need to understand how to use this `allowPodArgs`